node/resolvers/catalog/index.ts

```diff
@@ -41,6 +43,71 @@ interface SearchContextParams {
   subcategory: string
 }
 
+const getBrandFromSlug = async (brandSlug: string, {dataSources:{catalog}}: Context)  => {
+  const brands = await catalog.brands()
+  return <Brand>find(
+    compose(
+      equals(brandSlug),
+      toLower,
+      CatalogSlugify,
+      prop('name') as any
+    ),
+    brands
+  )
+}
+
+const findClusterNameFromId = (products: Product[], clusterId: string) => {
+  // Slice only the first fifty to prevent problem if this products array eventually get to thousands of units
+  const allSearchableClusters = 
+    products.slice(0, 50).reduce<Record<string, string>>((acc, prod) => {
+      const { searchableClusters } = prod
```

Comment:
should probably be `productClusters`, as one might create a direct link to a non-searchable cluster